File: snippets/fsharp/System/Object/ReferenceEquals/referenceequalsa.fs
module referenceequalsa

// <Snippet2>
open System

let s1 = "String1"
let s2 = "String1"
printfn $"s1 = s2: {Object.ReferenceEquals(s1, s2)}"
printfn $"""{s1} interned: {if String.IsNullOrEmpty(String.IsInterned s1) then "No" else "Yes"}"""

let suffix = "A"
let s3 = "String" + suffix
let s4 = "String" + suffix
printfn $"s3 = s4: {Object.ReferenceEquals(s3, s4)}"
printfn $"""{s3} interned: {if String.IsNullOrEmpty(String.IsInterned s3) then "No" else "Yes"}"""

// The example displays the following output:
//       s1 = s2: True
//       String1 interned: Yes
//       s3 = s4: False
//       StringA interned: No
// </Snippet2>

File: snippets/csharp/System/Object/ReferenceEquals/referenceequalsa.cs
using System;

public class ObjectReferenceEqualsExample2
{
    public static void Run()
    {
        // <Snippet2>
        string s1 = "String1";
        string s2 = "String1";
        Console.WriteLine($"s1 = s2: {object.ReferenceEquals(s1, s2)}");
        Console.WriteLine($"{s1} interned: {(string.IsNullOrEmpty(string.IsInterned(s1)) ? "No" : "Yes")}");

        string suffix = "A";
        string s3 = "String" + suffix;
        string s4 = "String" + suffix;
        Console.WriteLine($"s3 = s4: {object.ReferenceEquals(s3, s4)}");
        Console.WriteLine($"{s3} interned: {(string.IsNullOrEmpty(string.IsInterned(s3)) ? "No" : "Yes")}");

        // The example displays the following output:
        //       s1 = s2: True
        //       String1 interned: Yes
        //       s3 = s4: False
        //       StringA interned: No
        // </Snippet2>
    }
}
